src/main/java/com/synopsys/integration/alert/channel/event/NotificationToDistributionEventConverter.java

```diff
@@ -63,7 +67,11 @@ public NotificationToDistributionEventConverter(final DescriptorMap descriptorMa
     private DistributionEvent createChannelEvent(final CommonDistributionConfiguration config, final AggregateMessageContent messageContent) {
         final ChannelDescriptor channelDescriptor = descriptorMap.getChannelDescriptor(config.getChannelName());
         logger.info("Found descriptor {}", channelDescriptor.getName());
-        final DescriptorActionApi actionApi = channelDescriptor.getRestApi(ConfigContextEnum.DISTRIBUTION);
-        return actionApi.createChannelEvent(config, messageContent);
+        final Optional<DescriptorActionApi> actionApi = channelDescriptor.getActionApi(ConfigContextEnum.DISTRIBUTION);
```

Comment:
If this method returned an Optional<DistributionEvent> then lines 58 -60 could become a one line channelEvent.ifPresent(distributionEvents::add)

Reply:
```
    return channelDescriptor
        .getActionApi(ConfigContextEnum.DISTRIBUTION)
        .map(api -> api.createChannelEvent(config, messageContent));
```